lib/Model/Importer/Binary/MipsELFImporter.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "ELFImporter.h"
+
+using namespace llvm;
+
+template<typename T, bool HasAddend>
+class MipsELFImporter : public ELFImporter<T, HasAddend> {
+  // TODO: Use std::optional.
+  Optional<uint64_t> MIPSFirstGotSymbol;
+  Optional<uint64_t> MIPSLocalGotEntries;
+
+public:
+  MipsELFImporter(TupleTree<model::Binary> &Model,
+                  const object::ELFObjectFileBase &TheBinary,
+                  uint64_t PreferredBaseAddress) :
+    ELFImporter<T, HasAddend>(Model, TheBinary, PreferredBaseAddress),
+    MIPSFirstGotSymbol(None),
+    MIPSLocalGotEntries(None) {}
+
+  void parseDynamicTag(uint64_t Tag,
+                       MetaAddress Relocated,
+                       SmallVectorImpl<uint64_t> &NeededLibraryNameOffsets,
+                       uint64_t Val) override {
```

Comment:
Making all of `parseDynamicTag` virtual is a bit error prone, the implementer must make sure to invoke the base class' method.
Consider introducing a `virtual` `parseArchitectureSpecificDynamicTags` invoked by the (non-`virtual` `parseDynamicTag`),

Probably a similar solution would be beneficial for `import`.